stl/inc/numeric

```diff
@@ -619,9 +614,23 @@ _NODISCARD constexpr common_type_t<_Mt, _Nt> lcm(const _Mt _Mx, const _Nt _Nx) n
 template <class _Ty, enable_if_t<is_arithmetic_v<_Ty> && !is_same_v<remove_cv_t<_Ty>, bool>, int> = 0>
 _NODISCARD constexpr _Ty midpoint(const _Ty _Val1, const _Ty _Val2) noexcept {
     if constexpr (is_floating_point_v<_Ty>) {
+        if (_STD is_constant_evaluated()) {
+            if (_STD _Is_nan(_Val1)) {
+                return _Val1;
+            }
+
+            if (_STD _Is_nan(_Val2)) {
+                return _Val2;
+            }
+        } else {
+            if (_STD _Is_nan(_Val1) || _STD _Is_nan(_Val2)) {

```

Comment:
Before I purposely put this test below the high limit tests before the NaN tests because we expect NaNs to be uncommon; you're fixing `-fassociative-math` and friends but it seems like that would still be fixed if you put this in the old location?

Reply:
Performing the high limit test with `<=` raises underserved `FE_INVALID` when the input value is NaN.

Alternative approaches:
- Test with quiet comparison `islessequal`
- Test with `bit_cast` tricks
- Just ignore the issue of `FE_INVALID`

Reply:
I see, to raise floating point exceptions. Can you add a comment to places where you do seemingly needless operations to raise the right exceptions and add tests for that? Otherwise that's likely to break. (I understand that requires `/fp:except` which might be annoying to set up in our tests :( )

Reply:
Hmmm actually it looks like we already have an `/fp:except` flavor tested:

```
PM_CL="/EHsc /MDd /D_ITERATOR_DEBUG_LEVEL=2 /std:c++latest /fp:except /Zc:preprocessor"
```

so you should just be able to do that.